blog/content/post/2022-12-22-fobnail_application.md

```diff
@@ -0,0 +1,363 @@
+Attester (that is, the application used just for attestation, not provisioning)
+we can reduce the surface of attack. It also helps with rare cases when users
+inadvertently plugged in unprovisioned Token - in that case Token would also
+lit _all-good_ green LED in the end, making it indistinguishable from normal
+attestation. For the same reason we also gave up on the idea of unprovisioning
+the token by holding the button - neighboring USB device could potentially keep
+the button pressed
+
+Speaking of LEDs, we now have to steer them in a non-blocking manner. Simple
+delay loops were good enough when the Token was CoAP client, but now it has to
+be able to respond to the clients without making them wait too long, especially
+for longer signals like attestation result. We took advantage of that forced
+change to [expand and standardize **TBD**](TBD) blink codes produced by Fobnail
+Token.
+
+## Building and running
+
+Building hasn't change much since [last time](https://blog.3mdeb.com/2022/2022-05-25-fobnail_provisioning/#building-and-running).
+There are some changes done to produce `fobnail-attester-with-provisioning`
+along with non-provisioning version, but instructions for building didn't
+change.
+
+Despite changes in the architecture, applications are started mostly the same as
+before. Main difference is that the applications don't have to (and shouldn't)
+be already running when Fobnail Token is being plugged in. Attester changed a
+bit more, it now can take arguments which use Fobnail Token Services.
+Description of those can be obtained by starting Attester with `--help`:
+
+```
+Usage:
+    fobnail-attester [CMD]...
+
+Commands:
+    --read-file token_fname:local_fname
+    -r token_fname:local_fname
+        Read file token_fname from Fobnail Token and save it in local_fname.
+
+    --write-file local_fname:token_fname
+    -w local_fname:token_fname
+        Write file local_fname to token_fname on Fobnail Token.
+
+    --delete-file token_fname
+    -d token_fname
+        Remove file token_fname from Fobnail Token.
+
+All commands are executed only if the attestation was successful.
+
+Both --read-file and --write-file can take '-' as local_fname to use stdout
+and stdin, respectively. If '-' appears in multiple write commands, the same
+data is written to all files. If '-' appears in multiple read commands, output
+consists of concatenated content of all files, in the order in which they are
+read. Use './-' to access regular file named '-'.
+
+Multiple commands may be specified at once, in that case they are executed in
+order in which they appear on the command line. Attestation is performed only
+once. If any of the commands fails, further commands are not executed.
+```
+
+## Preparing encrypted disk image for use with Fobnail
+
+Some preparations have to be done during provisioning by administrator to make
+user's life as easy as possible.
+
+#### Provisioning
```

Comment:
We should let users know what OS version are we using and if it makes any difference what version of tools we use.